spec/unit/packages_inspector_spec.rb

```diff
@@ -31,68 +31,214 @@
+    end
 
-    packages_inspector.inspect(filter)
-    description.packages
-  end
+    def inspect_data(data = package_example)
+      expect(system).to receive(:run_command) { data }
 
-  it "returns a local SystemDescription containing rpm as one package" do
-    expect(inspect_data).to eq(expected_packages)
-  end
+      packages_inspector.inspect(filter)
+      description.packages
+    end
 
-  it "returns a remote SystemDescription containing rpm as one package" do
-    expect(inspect_data("remotehost", package_example)).to eq(expected_packages)
-  end
+    it "returns a local SystemDescription containing rpm as one package" do
+      expect(inspect_data).to eq(expected_packages)
+    end
 
-  it "ignores fake rpm packages with the name gpg-pubkey" do
-    data = "gpg-pubkey|39db7c82|1.0$#{package_example}"
-    expect(inspect_data(data)).to eq(expected_packages)
-  end
+    it "returns a remote SystemDescription containing rpm as one package" do
+      expect(inspect_data(package_example)).to eq(expected_packages)
+    end
+
+    it "ignores fake rpm packages with the name gpg-pubkey" do
+      data = "gpg-pubkey|39db7c82|1.0$\n#{package_example}"
+      expect(inspect_data(data)).to eq(expected_packages)
+    end
 
-  it "returns a summary" do
-    expect(system).to receive(:check_requirement) { true }
-    expect(system).to receive(:run_command) { package_example }
+    it "returns a summary" do
+      expect(system).to receive(:check_requirement) { true }
+      expect(system).to receive(:run_command) { package_example }
 
-    packages_inspector.inspect(filter)
-    expect(packages_inspector.summary).to include("Found 2 packages")
+      packages_inspector.inspect(filter)
+      expect(packages_inspector.summary).to include("Found 2 packages")
+    end
+
+    it "returns sorted data" do
+      names = inspect_data.map(&:name)
+      expect(names).to eq(names.sort)
+    end
   end
 
-  it "returns sorted data" do
-    names = inspect_data.map(&:name)
-    expect(names).to eq(names.sort)
+  context "dpkg inspection" do
+    let(:dpkg_output) {
+      <<EOF
+Desired=Unknown/Install/Remove/Purge/Hold
+| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
+|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
+||/ Name                                                   Version                          Architecture                     Description
++++-======================================================-================================-================================-==================================================================================================================
+ii  accountsservice:amd64                                  0:0.6.35-foo-0ubuntu7.2          amd64                            query and manipulate user account information
+ii  adduser                                                3.113+nmu3ubuntu3                all                              add and remove users and groups
```

Comment:
I think it would make sense to add one removed package to make sure a refactoring doesn't mess up the parsing. But of course it is not that important with the integration tests, but they might also have no removed packages.